Sources/Get/PackagesDirectory.swift

```diff
@@ -35,7 +34,7 @@ class PackagesDirectory {
 
         var result = Dictionary<String, Git.Repo>()
         for name in try! localFS.getDirectoryContents(self.prefix) {
-            let prefix = Path.join(self.prefix, name)
+            let prefix = self.prefix.appending(RelativePath(name))
```

Comment:
I agree that would be nice, though for symmetry we should then have `appending(subpath:)` for what we have now.

There is a bit of an ambiguity here with appending a subpath vs a name (which would presumably throw an error if the string contains a path separator).  That ambiguity can be dealt with through more verbosity, as you suggest, but one of the things I want is to be able to use operators to append paths.  This is because I think the extra text sometimes obscures the underlying intent.

I actually implemented all the diffs using `+` as a synonym for `appending()`, but decided to back off on that for this PR to avoid holding up the Path adoption on the inevitable resulting discussion.  That is something I don't want to give up on moving forward, though... the code is quite a bit more readable a operator for path concatenation.